src/sentry/static/sentry/app/components/hovercard.jsx

```diff
@@ -70,6 +79,71 @@ class Hovercard extends React.Component {
     return currentClasses.length > 0 ? currentClasses : classes;
   }
 
+  // figure out if we're dealing with hovercard-left or hovercard-right (in which case we don't
+  // want to center the hovercard) or hovercard or hovercard-bottom (in which case we do)
+  getMargin(classes) {
+    if (!this.cardElement || !this.state.visible) return {};
+
+    const rect = this.cardElement.getBoundingClientRect();
+
+    const rectWidth = this.isMultipleLines() ? rect.width : this.getSingleLineWidth();
+
+    if (classes.indexOf('left') !== -1 || classes.indexOf('right')) {
+      return -rectWidth / 2;
+    } else {
+      return 0;
+    }
+  }
+
+  isMultipleLines() {
+    if (!this.cardElement || !this.state.visible) return {};
+
+    // spans will create a new bounding box for each line of text
+    // if the text wraps, so this is a way to test for text wrapping
+    return this.cardBodySpan.getClientRects().length > 1;
+  }
+
+  //computed styles come back like "10px", so convert to the int 10
+  stylePxToNum(px) {
+    return parseInt(px.slice(0, -2), 10);
+  }
+
+  //this will be the width of the overall hovercard, based on the
+  //contents of the span in the body and all padding/borders around it
+  getSingleLineWidth() {
+    if (!this.cardElement || !this.state.visible) return {};
```

Comment:
since this returns an integer normally you should probably send over `null` or `0`.